src/helpers/normalizeContact.js

```diff
@@ -0,0 +1,79 @@
+import { models } from 'cozy-client'
+import log from 'cozy-logger'
+import { isEqual } from 'lodash'
+import { DOCTYPE_CONTACTS } from './doctypes'
+
+const {
+  getFullname,
+  getIndexByFamilyNameGivenNameEmailCozyUrl,
+  getDisplayName
+} = models.contact
+
+export const normalizeContact = contact => {
+  return {
+    ...contact,
+    fullname: getFullname(contact),
+    displayName: getDisplayName(contact),
+    indexes: {
+      byFamilyNameGivenNameEmailCozyUrl: getIndexByFamilyNameGivenNameEmailCozyUrl(
+        contact
+      )
+    }
+  }
+}
+
+export const fetchContactsToUpdate = async (client, lastSuccess) => {
+  try {
+    const lastSuccessUTCForced = new Date(lastSuccess).toISOString()
+    const contactsToUpdate = []
+
+    const queryDef = client
+      .find(DOCTYPE_CONTACTS)
+      .where({
+        trashed: {
+          $exists: false
+        },
+        $or: [
+          {
+            cozyMetadata: { $exists: false }
+          },
+          {
+            'cozyMetadata.updatedAt': { $gt: lastSuccessUTCForced }
+          }
+        ]
+      })
+      .indexFields(['_id'])
+      .limitBy(1000)
+
+    const result = await client.queryAll(queryDef)
+    const expected = result.map(contact => normalizeContact(contact))
+    result.map((contact, index) => {
+      if (!isEqual(contact, expected[index]))
+        return contactsToUpdate.push(contact)
+    })
+
+    log('info', `found ${contactsToUpdate.length} contact(s) to update`)
+
+    return contactsToUpdate
+  } catch (e) {
+    throw new Error(`Can't find elements : ${e}`)
+  }
+}
+
+export const fetchLastSuccessOfService = async (client, serviceName) => {
+  try {
+    const triggersResult = await client.query(
```

Comment:
This request is more about the `trigger` and the next one about `triggersResult` non ?

Reply:
the first one is all results according to the query (find by name). The second is a specific one according to its id. Named change for `triggersByName` and `trigger`. Good ?